### PR TITLE
refactor: Move virtual parent utilities to react-utilities package

### PR DESCRIPTION
**Pull request by ling1726**
Currently the only way to leverage virtual parents is to take a dependency on react-portal. In order for non-component packages (i.e. react-tabster) to use virtual parent utilties without depending on a component, the virtual parent utilties are moved to react-utilities in this case

deprecates the utilities already in @fluentui/react-portal

Fixes partially #28915

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

| Package & Exports                                                                             |      Baseline (minified/GZIP) |                            PR |                                                                                                Change |
| :-------------------------------------------------------------------------------------------- | ----------------------------: | ----------------------------: | ----------------------------------------------------------------------------------------------------: |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr> | `118.895 kB`<br />`37.406 kB` | `118.895 kB`<br />`37.414 kB` | <br />`8 B`  |

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                   |         Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------- | ---------------------------: |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContext.fixture.js'>createContext</abbr>                          |         `510 B`<br />`330 B` |
| <samp>global-context</samp> <br /> <abbr title='bundle-size/createContextSelector.fixture.js'>createContextSelector</abbr>          |         `537 B`<br />`342 B` |
| <samp>react-accordion</samp> <br /> <abbr title='bundle-size/Accordion.fixture.js'>Accordion (including children components)</abbr> | `91.932 kB`<br />`28.052 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/Badge.fixture.js'>Badge</abbr>                                             |  `25.987 kB`<br />`8.384 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/CounterBadge.fixture.js'>CounterBadge</abbr>                               |  `26.888 kB`<br />`8.687 kB` |
| <samp>react-badge</samp> <br /> <abbr title='bundle-size/PresenseBadge.fixture.js'>PresenceBadge</abbr>                             |  `24.393 kB`<br />`8.893 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/Button.fixture.js'>Button</abbr>                                          | `39.667 kB`<br />`10.794 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/CompoundButton.fixture.js'>CompoundButton</abbr>                          | `47.023 kB`<br />`12.295 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/MenuButton.fixture.js'>MenuButton</abbr>                                  | `44.466 kB`<br />`12.188 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/SplitButton.fixture.js'>SplitButton</abbr>                                | `52.469 kB`<br />`13.771 kB` |
| <samp>react-button</samp> <br /> <abbr title='bundle-size/ToggleButton.fixture.js'>ToggleButton</abbr>                              | `56.756 kB`<br />`12.702 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>          | `90.474 kB`<br />`27.605 kB` |
| <samp>react-divider</samp> <br /> <abbr title='bundle-size/Divider.fixture.js'>Divider</abbr>                                       |  `19.902 kB`<br />`7.415 kB` |
| <samp>react-field</samp> <br /> <abbr title='bundle-size/Field.fixture.js'>Field</abbr>                                             |   `21.61 kB`<br />`8.259 kB` |
| <samp>react-image</samp> <br /> <abbr title='bundle-size/Image.fixture.js'>Image</abbr>                                             |  `14.818 kB`<br />`5.902 kB` |
| <samp>react-jsx-runtime</samp> <br /> <abbr title='bundle-size/Classic.fixture.js'>Classic Pragma</abbr>                            |      `1.072 kB`<br />`544 B` |
| <samp>react-jsx-runtime</samp> <br /> <abbr title='bundle-size/DevRuntime.fixture.js'>JSX Dev Runtime</abbr>                        |   `2.893 kB`<br />`1.319 kB` |
| <samp>react-jsx-runtime</samp> <br /> <abbr title='bundle-size/Runtime.fixture.js'>JSX Runtime</abbr>                               |   `3.479 kB`<br />`1.554 kB` |
| <samp>react-label</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Label</abbr>                                             |  `13.225 kB`<br />`5.438 kB` |
| <samp>react-link</samp> <br /> <abbr title='bundle-size/Label.fixture.js'>Link</abbr>                                               |   `16.095 kB`<br />`6.53 kB` |
| <samp>react-overflow</samp> <br /> <abbr title='bundle-size/hooks.fixture.js'>hooks only</abbr>                                     |  `12.594 kB`<br />`4.702 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                          |  `12.362 kB`<br />`4.543 kB` |
| <samp>react-positioning</samp> <br /> <abbr title='bundle-size/usePopper.fixture.js'>usePositioning</abbr>                          |  `25.245 kB`<br />`9.141 kB` |
| <samp>react-provider</samp> <br /> <abbr title='bundle-size/FluentProvider.fixture.js'>FluentProvider</abbr>                        |  `21.511 kB`<br />`7.994 kB` |
| <samp>react-spinner</samp> <br /> <abbr title='bundle-size/Spinner.fixture.js'>Spinner</abbr>                                       |  `22.481 kB`<br />`8.146 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Default.fixture.js'>Text - Default</abbr>                              |  `15.832 kB`<br />`6.254 kB` |
| <samp>react-text</samp> <br /> <abbr title='bundle-size/Text.Wrappers.fixture.js'>Text - Wrappers</abbr>                            |   `19.01 kB`<br />`6.578 kB` |
| <samp>react-toast</samp> <br /> <abbr title='bundle-size/Toast.fixture.js'>Toast (including Toaster)</abbr>                         | `91.789 kB`<br />`27.423 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                       | `51.348 kB`<br />`18.113 kB` |
| <samp>react-utilities</samp> <br /> <abbr title='bundle-size/SSRProvider.fixture.js'>SSRProvider</abbr>                             |         `180 B`<br />`159 B` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/5b6536c989c9ebb2bf4010fbf18dfb50b570b2c6'>5b6536c989c9ebb2bf4010fbf18dfb50b570b2c6</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/Avatar-mount_base.html">631</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/Avatar-mount.html">632</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/Button-mount_base.html">326</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/Button-mount.html">309</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/Field-mount_base.html">1147</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/Field-mount.html">1169</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/FluentProvider-mount_base.html">713</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/FluentProvider-mount.html">703</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/FluentProviderWithTheme-mount_base.html">91</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/FluentProviderWithTheme-mount.html">81</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">74</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">72</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">80</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>InfoButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/InfoButton-mount_base.html">7</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/InfoButton-mount.html">13</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/MakeStyles-mount_base.html">864</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/MakeStyles-mount.html">865</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/Persona-mount_base.html">1746</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/Persona-mount.html">1715</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/SpinButton-mount_base.html">1416</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/29286/perf-test-react-components/SpinButton-mount.html">1371</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 🕵 fluentuiv9 No visual regressions between this PR and main
<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

File: packages/react-components/react-portal/etc/react-portal.api.md [hunk 1]
```diff
@@ -6,9 +6,6 @@
 
 import * as React_2 from 'react';
 
-// @public
```
Comment (sopranopillow) [inline packages/react-components/react-portal/etc/react-portal.api.md:9]: I think you might need to generate the api since this hasn't updated

**Review by sopranopillow — APPROVED** [review]
Other than the api file comment, looks great!

File: packages/react-components/react-portal/src/virtualParent/getParent.ts [hunk 2]
```diff
@@ -3,12 +3,15 @@ import { isVirtualElement } from './isVirtualElement';
 type GetParentOptions = {
   /**
    * Indicates if getParent() should ignore a virtual parent.
+   * @internal
```
Comment (layershifter) [inline packages/react-components/react-portal/src/virtualParent/getParent.ts:6]: Please re-export utils instead of just keeping them.

Comment (ling1726) [reply packages/react-components/react-portal/src/virtualParent/getParent.ts:6]: done

**Pull request merged**